File: 1381-design-a-stack-with-increment/CustomStack.java
import java.util.Arrays;

class CustomStack {

    int[] stack;
    int index = -1;

    public CustomStack(int maxSize) {
        stack = new int[maxSize];
    }
    
    public void push(int x) {
        if(index < stack.length - 1) {
            stack[++index] = x;
        }
    }
    
    public int pop() {
        if(index == -1) return -1;
        return stack[index--];
    }
    
    public void increment(int k, int val) {
        int size = Math.min(index, k - 1);
        for(int i = 0; i <= size; i++) {
            stack[i] += val; 
        }
    }

    @Override
    public String toString() {
        return Arrays.toString(stack) + ", i = " + index;
    }
}

/**
 * Your CustomStack object will be instantiated and called as such:
 * CustomStack obj = new CustomStack(maxSize);
 * obj.push(x);
 * int param_2 = obj.pop();
 * obj.increment(k,val);
 */

File: 1381-design-a-stack-with-increment/CustomStack.py
class CustomStack:

    def __init__(self, maxSize: int):
        self.stack = [0] * maxSize
        self.index = -1

    def push(self, x: int) -> None:
        if self.index < len(self.stack) - 1:
           self.index += 1
           self.stack[self.index] = x 
        

    def pop(self) -> int:
        if self.index < 0:
            return -1
        
        val = self.stack[self.index]
        self.index -= 1
        return val
        

    def increment(self, k: int, val: int) -> None:
        size = min(self.index, k - 1)

        for i in range(size + 1):
            self.stack[i] += val
        
stack = CustomStack(3)
stack.push(1)
stack.push(2)

print(stack.stack, stack.index)

stack.pop()
stack.push(2)
stack.push(3)
stack.push(4)

print(stack.stack, stack.index)

stack.increment(5, 100)
stack.increment(2, 100)

print(stack.stack, stack.index)

stack.pop()
stack.pop()
stack.pop()
stack.pop()

print(stack.stack, stack.index)

# Your CustomStack object will be instantiated and called as such:
# obj = CustomStack(maxSize)
# obj.push(x)
# param_2 = obj.pop()
# obj.increment(k,val)